tracer/src/Datadog.Trace/Tagging/TraceSourcesUtils.cs

```diff
@@ -0,0 +1,84 @@
+// <copyright file="TraceSourcesUtils.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Datadog.Trace.Tagging
+{
+    /// <summary>
+    /// Utility funtions to handle TraceSources values in the '_dd.p.ts' tag
+    /// </summary>
+    internal static class TraceSourcesUtils
+    {
+        public static TraceSources GetTraceSources(this TraceTagCollection? tags)
+        {
+            if (tags == null)
+            {
+                return TraceSources.None;
+            }
+
+            return GetTraceSources(tags.GetTag(Tags.Propagated.TraceSource));
+        }
+
+        public static TraceTagCollection? EnableTraceSources(this TraceTagCollection? tags, TraceSources sources)
+        {
+            if (tags != null)
+            {
+                var value = GetTraceSources(tags.GetTag(Tags.Propagated.TraceSource)) | sources;
+                if (value != TraceSources.None)
+                {
+                    tags.SetTag(Tags.Propagated.TraceSource, ToTag(value));
```

Comment:
Should we make this get/set "thread-safe"?

Reply:
You mean make the operation atomic? We could lock the section, but the perf penalty does not justify the benefits. I doubt there will be multiple threads setting this value at once (or at all)